samcli/lib/build/app_builder.py

```diff
@@ -216,20 +216,26 @@ def update_template(template_dict: Dict, original_template_path: str, built_arti
 
             resource_type = resource.get("Type")
             properties = resource.setdefault("Properties", {})
+            package_type = properties.get("PackageType", ZIP)
 
-            if resource_type == SamBaseProvider.SERVERLESS_FUNCTION and properties.get("PackageType", ZIP) == ZIP:
-                properties["CodeUri"] = store_path
+            if resource_type == SamBaseProvider.SERVERLESS_FUNCTION and package_type == ZIP:
+                if "InlineCode" not in properties or properties["InlineCode"] is None:
+                    properties["CodeUri"] = store_path
 
-            if resource_type == SamBaseProvider.LAMBDA_FUNCTION and properties.get("PackageType", ZIP) == ZIP:
-                properties["Code"] = store_path
+            if resource_type == SamBaseProvider.LAMBDA_FUNCTION and package_type == ZIP:
+                if "Code" in properties:
```

Comment:
I think you should be able to do it all in one condition:
```python
if "Code" not in properties or "ZipFile" not in properties["Code"] or properties["Code"]["ZipFile"] is None:
```

Reply:
thanks ;) separated the conditions because they are two different cases, but I guess merging them is easy to understand as well